numpy/core/src/multiarray/item_selection.c

```diff
@@ -2093,13 +2093,17 @@ PyArray_CountNonzero(PyArrayObject *self)
         PyArray_PREPARE_TRIVIAL_ITERATION(self, count, data, stride);
 
         while (count--) {
-            if (nonzero(data, self)) {
+            npy_bool n = nonzero(data, self);
+            if (n == -1) {
```

Comment:
Ugh you're right, I should have caught that. `npy_bool` is `unsigned char`.

I was trying to avoid calling `PyErr_Occurred` every time through the loop. My timing tests suggested that gives a 40% slowdown.